src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java

```diff
@@ -229,6 +232,21 @@ public void checkLetterInBulkPrint(String jurorNumber, String formType, LocalDat
         }
     }
 
+    @Override
+    public void removeQueuedLetterForJuror(JurorPool jurorPool) {
+
+        // determine form codes for juror status
+        List<FormCode> formCodes = FormCode.getFormCodesForJurorStatus(jurorPool.getStatus().getStatus());
+
+        List<BulkPrintData> bulkPrintDataList = bulkPrintDataRepository
+            .findByJurorNoAndCreationDateAndFormAttributeFormTypeIn(
+            jurorPool.getJuror().getJurorNumber(), LocalDate.now(),
+            formCodes.stream().map(FormCode::getCode).toList()
+        );
+
+        bulkPrintDataRepository.deleteAll(bulkPrintDataList);
```

Comment:
We should already strip out letters that do not need to be printed apart of the letter Job.

I.e we check juror status and if they have changed we skip that row for printing

Reply:
understood, I think its cleaner to remove the queued letter as its not required at all